tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+            const same_as<iterator_t<R>> auto i2 = r2.begin();
+            if (!is_empty) {
+                assert(*i2 == *i);
+                assert(*r2.begin() == *i2);
+                assert(*r2.begin() == *i2);
+            }
+        }
+
+        const same_as<iterator_t<const R>> auto ic = as_const(r).begin();
+        if (!is_empty) {
+            assert(*ic == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto rc2                                    = as_const(r);
+            const same_as<iterator_t<const R>> auto ic2 = rc2.begin();
+            if (!is_empty) {
+                assert(*ic2 == *ic);
+                assert(*rc2.begin() == *ic2);
+                assert(*rc2.begin() == *ic2);
+            }
+        }
+    }
+
+    // Validate take_while_view::end
+    STATIC_ASSERT(CanMemberEnd<R>);
+    STATIC_ASSERT(CanMemberEnd<const R>);
+    STATIC_ASSERT(CanEnd<const R&> == ranges::range<const V>);
+    STATIC_ASSERT(!common_range<R>);
+    STATIC_ASSERT(!common_range<const R>);
+    if (!is_empty) {
+        [[maybe_unused]] same_as<ranges::sentinel_t<R>> auto s        = r.end();
+        [[maybe_unused]] same_as<ranges::sentinel_t<const R>> auto sc = as_const(r).end();
+
+        if (forward_range<V>) { // intentionally not if constexpr
+            // Compare with const / non const iterators
+            assert(s != r.begin());
+            assert(s != as_const(r).begin());
+            assert(sc != r.begin());
+            assert(sc != as_const(r).begin());
+
+            // Compare with end of range
+            if constexpr (common_range<V>) {
+                assert(s == s.base());
+                assert(s == sc.base());
+                assert(sc == s.base());
+                assert(sc == sc.base());
+            } else {
+                if constexpr (forward_range<V>) {
+                    const auto length = 8; // NB: depends on the test data
+                    assert(s == next(r.begin(), length));
+                    assert(s == next(as_const(r).begin(), length));
+                    assert(sc == next(r.begin(), length));
+                    assert(sc == next(as_const(r).begin(), length));
+                }
+            }
+
+            // Compare with iterator whose predicate evaluates to false
+            if constexpr (forward_range<V>) {
+                const auto length = 4; // NB: depends on the test data
+                assert(s == next(r.begin(), length));
+                assert(s == next(as_const(r).begin(), length));
+                assert(sc == next(r.begin(), length));
+                assert(sc == next(as_const(r).begin(), length));
+            }
+        }
+    }
+
+    // Validate view_interface::data
+    STATIC_ASSERT(CanMemberData<R> == contiguous_range<V>);
+    STATIC_ASSERT(CanData<R&> == contiguous_range<V>);

```

Comment:
`contiguous_range<V>` implies `CanMemberData<R>` which implies `CanData<R&>`, so this is somewhat redundant.

Reply:
Removed